vsphere_plugin_common/__init__.py

```diff
@@ -1815,11 +1875,45 @@ def create_server(
                 "Error during executing VM creation task. VM name: \'{0}\'."
                 .format(vm_name))
 
-        vm = self._get_obj_by_name(
+        # VM object created. Now perform final post-creation tasks
+        vm = task.info.result
+        logger().info('VM version: vmx-{old}/vmx-{new}'.format(
```

Comment:
Never noticed this before? Why do we instantiate logger again each time we have a message?

Reply:
I have changed code to get logger in constructor.